eclipse-scout-core/src/popup/Popup.js

```diff
@@ -383,9 +390,97 @@ export default class Popup extends Widget {
+    } else {
+      widgets.resetProperty(v => this.setWithGlassPane(v), this, '_withGlassPane');
+      widgets.resetProperty(v => this.setCloseOnAnchorMouseDown(v), this, '_closeOnAnchorMouseDown');
+      widgets.resetProperty(v => this.setCloseOnMouseDownOutside(v), this, '_closeOnMouseDownOutside');
+      widgets.resetProperty(v => this.setCloseOnOtherPopupOpen(v), this, '_closeOnOtherPopupOpen');
+    }
+  }
+
+  _renderModal() {
+    this.$container.toggleClass('modal', this.modal);
+  }
+
+  setWithGlassPane(withGlassPane) {
+    if (!this.modal) {
+      this.setProperty('withGlassPane', withGlassPane);
+    } else {
+      this._withGlassPane = withGlassPane;
+    }
+  }
+
   _renderWithGlassPane() {
-    if (this._glassPaneRenderer) {
+    if (this.withGlassPane && !this._glassPaneRenderer) {
+      this._glassPaneRenderer = new GlassPaneRenderer(this);
       this._glassPaneRenderer.renderGlassPanes();
+    } else if (!this.withGlassPane && this._glassPaneRenderer) {
+      this._glassPaneRenderer.removeGlassPanes();
+      this._glassPaneRenderer = null;
+    }
+  }
+
+  setCloseOnMouseDownOutside(closeOnMouseDownOutside) {
+    if (!this.modal) {
+      this.setProperty('closeOnMouseDownOutside', closeOnMouseDownOutside);
+    } else {
+      this._closeOnMouseDownOutside = closeOnMouseDownOutside;
+    }
+  }
+
+  _renderCloseOnMouseDownOutside() {
+    // The listener needs to be executed in the capturing phase -> prevents that _onDocumentMouseDown will be executed right after the popup gets opened using mouse down, otherwise the popup would be closed immediately
+    if (this.closeOnMouseDownOutside) {
+      if (!this._documentMouseDownHandler) {
+        this._documentMouseDownHandler = this._onDocumentMouseDown.bind(this);
+        this.$container.document(true).addEventListener('mousedown', this._documentMouseDownHandler, true); // true=the event handler is executed in the capturing phase
+      }
+    } else {
+      if (this._documentMouseDownHandler) {
+        this.$container.document(true).removeEventListener('mousedown', this._documentMouseDownHandler, true);
+        this._documentMouseDownHandler = null;
+      }
+    }
+  }
+
+  setCloseOnAnchorMouseDown(closeOnAnchorMouseDown) {
+    if (!this.modal) {
+      this.setProperty('closeOnAnchorMouseDown', closeOnAnchorMouseDown);
+    } else {
+      this._closeOnAnchorMouseDown = closeOnAnchorMouseDown;
+    }
+  }
+
+  setCloseOnOtherPopupOpen(closeOnOtherPopupOpen) {
+    if (!this.modal) {
+      this.setProperty('closeOnOtherPopupOpen', closeOnOtherPopupOpen);
+    } else {
+      this._closeOnOtherPopupOpen = closeOnOtherPopupOpen;
+    }
+  }
+
+  _renderCloseOnOtherPopupOpen() {
+    if (this.closeOnOtherPopupOpen) {
+      if (!this._popupOpenHandler) {
+        this._popupOpenHandler = this._onPopupOpen.bind(this);
+        this.session.desktop.on('popupOpen', this._popupOpenHandler);
+      } else {
+        if (this._popupOpenHandler) {
```

Comment:
There is a bracket error: else should probably refer to the outer if not the inner one.
Maybe it would be clearer if the if statements were merged like in _renderWithGlassPane